packages/gatsby-transformer-remark/src/extend-node-type.js

```diff
@@ -435,9 +459,27 @@ module.exports = (
             type: GraphQLString,
             defaultValue: `fields.slug`,
           },
+          maxDepth: {
+            type: GraphQLInt,
+            defaultValue: 6,
+          },
+          heading: {
+            type: GraphQLString,
+            defaultValue: null,
+          },
+          tight: {
+            type: GraphQLBoolean,
+            defaultValue: false,
+          },
         },
-        resolve(markdownNode, { pathToSlugField }) {
-          return getTableOfContents(markdownNode, pathToSlugField)
+        resolve(markdownNode, { pathToSlugField, maxDepth, heading, tight }) {
```

Comment:
I would just pass `args` (without destructuring it here) as is to `getTableOfContents` function and destructure it there

Reply:
fixed in new commit

Reply:
```suggestion
        resolve(markdownNode, args) {
```